web-common/src/components/forms/MultiInput.svelte

```diff
@@ -0,0 +1,116 @@
+<script lang="ts">
+  import { IconButton } from "@rilldata/web-common/components/button";
+  import InfoCircle from "@rilldata/web-common/components/icons/InfoCircle.svelte";
+  import { slide } from "svelte/transition";
+
+  /**
+   * Input that allows to enter multiple items but appears within a single input box.
+   * This is a more advanced version of InputArray.svelte
+   */
+  import Tooltip from "@rilldata/web-common/components/tooltip/Tooltip.svelte";
+  import TooltipContent from "@rilldata/web-common/components/tooltip/TooltipContent.svelte";
+  import { XIcon } from "lucide-svelte";
+  import type { createForm } from "svelte-forms-lib";
+
+  export let id: string;
+  export let label = "";
+  export let description = "";
+  export let accessorKey: string;
+  export let hint = "";
+
+  export let formState: ReturnType<typeof createForm<Record<string, any>>>;
+  const { form, errors, validateField } = formState;
+  $: values = $form[id] as Record<string, string>[];
+  // There's a bug in how `svelte-forms-lib` types the `$errors` store for arrays.
+  // See: https://github.com/tjinauyeung/svelte-forms-lib/issues/154#issuecomment-1087331250
+  $: errs = ($errors[id] as unknown as Record<string, string>[]) ?? [];
```

Comment:
We would have to duplicate the code to add and remove in the caller, since `$form[id] = $form[id].concat...` has to be done to actually update the array. Perhaps we can revisit this once we move to the new form lib?